source/dagger/gameplay/team_game/game_manager.cpp

```diff
@@ -0,0 +1,99 @@
+#include "game_manager.h"
+
+#include <core/engine.h>
+#include <core/graphics/sprite.h>
+
+using namespace dagger;
+using namespace team_game;
+
+void GameManagerSystem::SpinUp()
+{
+    Engine::Dispatcher().sink<NextFrame>().connect<&GameManagerSystem::OnEndOfFrame>(this);
+}
+
+void GameManagerSystem::WindDown()
+{
+    Engine::Dispatcher().sink<NextFrame>().disconnect<&GameManagerSystem::OnEndOfFrame>(this);
+}
+
+void GameManagerSystem::Run()
+{
+    auto view = Engine::Registry().view<Level>();
+    for (auto entity : view)
+    {
+        auto& level = view.get<Level>(entity);
+        UInt8 currentLevel = level.level;
+
+        if (level.completedObjective)
+        {
+            auto& level = Engine::Registry().get_or_emplace<Level>(entity);
+            level.level = currentLevel;
+            level.completedObjective = false;
+
+            Engine::Registry().view<ComponentOfLevel>().each([](ComponentOfLevel& levelComponent_) 
+            {
+                levelComponent_.discard = true;
+            });
+
+            LoadNextLevel(level);
+        }
+    }
+}
+
+void GameManagerSystem::LoadNextLevel(Level& level_)
+{
+    level_.level += 1;
+    String levelName = fmt::format("levels/level_{}.txt", level_.level);
+    FilePath path{ levelName };
+    std::ifstream fin(Files::absolute(path).string().c_str());
+
+    Vector2 pivotGlobal = { 400.0f, 300.0f };
+
+    String baseDir, textureName;
+    Float32 zPos, horizontalBlocks, verticalBlocks;
+    Vector2 blockSize, scale, bias;
+
+    auto& reg = Engine::Registry();
+
+    while (fin >> baseDir >> textureName >> zPos >>
+        blockSize.x >> blockSize.y >>
+        scale.x >> scale.y >> bias.x >> bias.y >>
+        horizontalBlocks >> verticalBlocks)
+    {
+        for (int i = 0; i < horizontalBlocks; i++)
+        {
+            for (int j = 0; j < verticalBlocks; j++)
+            {
+                auto block = reg.create();
+                auto& spriteBlock = reg.get_or_emplace<Sprite>(block);
+                auto& componentOfLevel = reg.get_or_emplace<ComponentOfLevel>(block);
+
+                String rootDir = "TeamGame:";
+                AssignSpriteTexture(spriteBlock, rootDir + baseDir + ":" + textureName);
+                spriteBlock.position = { i * blockSize.x - pivotGlobal.x + blockSize.x / 2 + bias.x,
+                                         j * blockSize.y - pivotGlobal.y + blockSize.y / 2 + bias.y,
+                                         zPos };
+                spriteBlock.scale = scale;
+            }
+        }
+    }
+}
+
+void GameManagerSystem::OnEndOfFrame()
+{
+    auto levelComponentsView = Engine::Registry().view<ComponentOfLevel>();
+
+    for (auto& entity : levelComponentsView)
+    {
+        auto& levelComponent = levelComponentsView.get<ComponentOfLevel>(entity);
+        if (levelComponent.discard)
+        {
+            Engine::Registry().remove_all(entity);
```

Comment:
my suggestion is to do complete removal here, and load another level after that.
(check racing reset for example)

that way it'll be easier to clean everything properly and speed up the removal process. 

You might need to save some player states or something between levels. But I think you will be able to do that even with full entities reset.

++ @gavrilovmiroslav